### PR TITLE
Fix enh-ruby-toggle-block edge cases / fix tests

### LINKED ISSUE #132: ruby-toggle-block/brace fails randomly even in isolation
`rake docker` runs all the tests in a completely isolated environment and this tests fails ~50% of the time. I have yet to dig into why (but have upgraded all of the docker infrastructure to make sure it isn't a versioning thing).

### PR DESCRIPTION
**Pull request by thiagoa**
With a hash inside a brace-delimited block, `enh-ruby-toggle-block` would mistake the hash for the block and give incorrect results. This PR makes `enh-ruby-toggle-block` leverage text properties to determine where the block is, and tries to improve code clarity.

## Old behavior

Before:

```ruby
    let(:dont_let) { { a: 'a', b: 'b' } }
```

After:

```ruby
    let(:dont_let) { do
        a: 'a', b: 'b' 
      end }
```

## New behavior

After:

```ruby
    let(:dont_let) do
      { a: 'a', b: 'b' } 
    end
```

It also fixes skipped tests for `enh-ruby-toggle-block` by waiting for erm to parse the buffer and for fontification to happen before running expectations.

Closes #132

### CODE REVIEW

File: enh-ruby-mode.el [hunk 1]
```diff
@@ -741,20 +741,21 @@ Warning: does not play well with command ‘electric-indent-mode’."
...
-            (if (match-beginning 1)
-                (enh-ruby-brace-to-do-end beg end)
-              (enh-ruby-do-end-to-brace beg end)))
-      (goto-char start))))
+  (let* ((pos (point))
+         (block-start (save-excursion
+                        (end-of-line)
+                        (while (and (not (bobp))
+                                    (not (enh-ruby-point-block-p)))
+                          (backward-char))
```
Comment (thiagoa) [inline enh-ruby-mode.el]: Not sure if this is the most performant way to search backward for a block using syntax properties. This function would _probably_ be slow in 0.1% of the cases when dealing with giant blocks or giant files. But it would work very well for the majority of use cases.

**Comment by zenspider** [comment]
Huh. I pulled your commits directly into master with `git am` but github didn't detect it. So, this is done. I changed the tests a bit afterwards. Thank you, and sorry it took me a while to get to it.
